src/core/search/indices.cc

```diff
@@ -560,34 +564,45 @@ struct HnswlibAdapter {
                100 /* seed*/} {
   }
 
-  void Add(const float* data, DocId id) {
-    if (world_.cur_element_count + 1 >= world_.max_elements_)
-      world_.resizeIndex(world_.cur_element_count * 2);
-    world_.addPoint(data, id);
+  void Add(const float* data, GlobalDocId id) {
+    while (true) {
+      try {
+        absl::ReaderMutexLock lock(&resize_mutex_);
```

Comment:
please add a comment that resizes needs to be protected and add the link to the issue.